ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineContext.java

```diff
@@ -208,16 +231,90 @@ DdlCommand createDdlCommand(
   String executeDdl(
       final String sqlExpression,
       final DdlCommand command,
-      final boolean withQuery
+      final boolean withQuery,
+      final Set<SourceName> withQuerySources
   ) {
-    final DdlCommandResult result = ddlCommandExec.execute(sqlExpression, command, withQuery);
+    if (command instanceof DropSourceCommand) {
+      throwIfOtherQueriesExist(((DropSourceCommand) command).getSourceName());
+    }
```

Comment:
is there any way to avoid special casing `DropSourceCommand` and just passing what we need into the executor? (applies below as well)

Reply:
This  casing also bothers me, but I the only way I found is to pass the `EngineContext` to the `DdlCommandExec`, and do the checks when dropping the source inside, even stop the query inside too. It requires that `DdlCommandExec` know queries, so I wasn't sure if we wanted to do that there. I left it here as queries is in the engine context.

If you like the idea of passing the EngineContext to the DdlCommandExec, then I can do it in a follow-up PR for 0.15. What do you think?

Reply:
makes sense, let's keep it as is for now and we can explore a refactor another time if this bites us